src/pages/workspace/WorkspaceInviteMessagePage.js

```diff
@@ -71,180 +71,144 @@ const defaultProps = {
     invitedEmailsToAccountIDsDraft: {},
 };
 
-class WorkspaceInviteMessagePage extends React.Component {
-    constructor(props) {
-        super(props);
+function WorkspaceInviteMessagePage(props) {
+    const focusTimeoutRef = useRef();
+    const welcomeMessageInputRef = useRef(null);
 
-        this.sendInvitation = this.sendInvitation.bind(this);
-        this.validate = this.validate.bind(this);
-        this.openPrivacyURL = this.openPrivacyURL.bind(this);
-        this.debouncedSaveDraf = _.debounce((newDraft) => {
-            Policy.setWorkspaceInviteMessageDraft(this.props.route.params.policyID, newDraft);
-        }, 2000);
-        this.state = {
-            welcomeNote: this.props.workspaceInviteMessageDraft || this.getDefaultWelcomeNote(),
-        };
-    }
-
-    componentDidMount() {
-        if (_.isEmpty(this.props.invitedEmailsToAccountIDsDraft)) {
-            Navigation.goBack(ROUTES.WORKSPACE_INVITE.getRoute(this.props.route.params.policyID), true);
-            return;
-        }
-        this.focusWelcomeMessageInput();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (!prevProps.isFocused && this.props.isFocused) {
-            this.focusWelcomeMessageInput();
-        }
+    const focusWelcomeMessageInput = () => {
+        focusTimeoutRef.current = setTimeout(() => {
+            welcomeMessageInputRef.current.focus();
+            // Below condition is needed for web, desktop and mweb only, for native cursor is set at end by default.
+            if (welcomeMessageInputRef.current.value && welcomeMessageInputRef.current.setSelectionRange) {
+                const length = welcomeMessageInputRef.current.value.length;
+                welcomeMessageInputRef.current.setSelectionRange(length, length);
+            }
+        }, CONST.ANIMATED_TRANSITION);
+    };
```

Comment:
You can consider using [`useAutofocusInput`](https://github.com/Expensify/App/blob/main/src/hooks/useAutoFocusInput.js) instead.

Reply:
Swapped current implementation for `useAutoFocusInput`